components/edit_channel_purpose_modal/edit_channel_purpose_modal.tsx

```diff
@@ -102,24 +91,23 @@ class EditChannelPurposeModal extends React.PureComponent {
         }
 
         this.setState({requestStarted: true});
-        const {data, error} = await patchChannel(channel.id, {purpose});
+        const result = await patchChannel(channel.id, {...channel, purpose});
```

Comment:
Looking at the way we use this function, it looks like the second argument of `patchChannel` should be `Partial<Channel>` so that we can just send up the changes that the patch requires.

We'll likely need to change this above, and in the original definition in `packages/mattermost-redux`

Reply:
This should be resolved, the redux pr to make this same change can be found here: mattermost/mattermost-redux#1424